### ISSUE TITLE
BTRFS filesystem requires more than 30MB (KIWI default minimum)

### ISSUE THREAD

**Issue by davidcassany** [OSInside/kiwi#1835]
## Problem description

If building an OEM expandable image including btrfs LVM volumes, the minimal size used for those volumes turns to be 30MB (KIWI default). However BTRFS refuses to format a filesystem that is less than 128MB.

This can be reproduced with something like:
```xml
<systemdisk name="onit" preferlvm="true">
        <volume name="home"/>
        <volume name="root" size="5G" />
        <volume name="standby" size="5G" label="STANDBY"/>
        <volume name="tmp"/>
        <volume name="opt"/>
        <volume name="srv"/>
        <volume name="boot/grub2/i386-pc"/>
        <volume name="boot/grub2/x86_64-efi" mountpoint="boot/grub2/x86_64-efi"/>
        <volume name="usr/local"/>
        <volume name="var" copy_on_write="false" size="5G" label="VAR" mountpoint="/var"/>
        <volume name="containerd" copy_on_write="false" size="1G" label="CONTAINERD" mountpoint="/var/lib/rancher/k3s/containerd"/>
        <volume name="storage" copy_on_write="false" size="1G" label="STORAGE" mountpoint="/var/lib/rancher/k3s/storage"/>
        <volume name="rancherconf" copy_on_write="false" size="1G" label="RANCHERCONF" mountpoint="/etc/rancher"/>
    </systemdisk>
```

Producing an error similar to:
```
[ INFO    ]: 07:56:13 | Creating volumes(btrfs)
[ INFO    ]: 07:56:13 | --> volume home with 30 MB
[ ERROR   ]: 07:56:13 | KiwiCommandError: mkfs.btrfs: stderr: ERROR: '/dev/onit/home' is too small to make a usable filesystem
ERROR: minimum size for each btrfs device is 114294784
, stdout: btrfs-progs v4.19.1
```

## Expected behaviour

Volumes creation does not fail.

## OS and Software information
+ KIWI version: 
+ Operating system: 
+ OBS version:

**Comment by schaefi** [comment]
yeah we know btrfs is space hungry. The 30MB is a min size setting for a volume without a size setup. The setting is applied in addition to the data blocks of data which is found in the image root tree for this volume. These default setting works for any filesystem we have ... except btrfs.

As it's also not possible to pre calculate the size of a btrfs filesystem to hold X bytes of real blocks, there is no clean way for
a good default or size calculation. This is the reason why there is no extra handling of size for the btrfs filesystem and the only good answer I can give here is: specify a size :-)

All empiric methods I tried in the past failed for btrfs under certain circumstances

From a coding perspective we can have a conversation to maintain different volume
default sizes per volume management.

```python
@staticmethod
def get_min_volume_mbytes():
    """
    Provides default minimum volume size in mbytes

    :return: mbsize value

    :rtype: int
    """
    return 30
```

This method is used in LVM and btrfs context

I'm fine to have this more granular, but I very much doubt it will really fix the size issue for btrfs filesystems

**Comment by davidcassany** [comment]
> the only good answer I can give here is: specify a size :-)

Yes indeed, this is the same answer I gave on the chat conversation and apparently it did not work out. That's when I decided to fill the issue, the work around that actually worked was to include some dummy file of 128MB inside the offending volume. The problem of the above setup is that it is a really niche case, using LVM on top of btrfs is not so common and even less common in KIWI. However the description is a valid one. I have to admit I did not test myself this setup, however we discussed it with some user in the chat long and deep enough to make me think something unexpected was happening on KIWI side.

**Comment by Conan-Kudo** [comment]
Why not just raise the minimum size?

**Comment by schaefi** [comment]
raise to what value to make btrfs happy ? This is non deterministic. I would rather go with David's suggestion and add an extra default size handling for btrfs in kiwi. That's what all the others did as well ```btrfs filesystem df``` vs. df ... same story

**Comment by Conan-Kudo** [comment]
Sure, I guess. It makes sense to have that capability anyway.

**Comment by schaefi** [comment]
From the community meeting we agreed on:

* provide one minimum default size setting for volumes
* extend the interface for filesystems to allow setting a constraint for the minimum size
* apply minimum size constraint to the btrfs filesystem from which we know the current default is not sufficient

**Comment by Conan-Kudo** [comment]
Uhh did we ever do this? We still need to ensure Btrfs has a minimum size of 16MiB when using `mkfs.btrfs --mixed` and 109 MiB otherwise.

**Comment by schaefi** [comment]
when using btrfs the safe default without any size is set to 120 MB. The value can be overwritten
